app/models.py

```diff
@@ -0,0 +1,81 @@
+from typing import Optional
+from uuid import UUID, uuid4
+from datetime import datetime, timedelta, timezone
+from jose import JWTError, jwt
+from pydantic import BaseModel, EmailStr, Field
+from passlib.context import CryptContext
+
+#Set up password hashing context using bcrypt
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+#JWT settings
+SECRET_KEY = "your_secret_key"
+ALGORITHM = "HS256"
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+
+
+class UserBase(BaseModel):
+    username: str
+    email: EmailStr
+    is_admin: bool = False
+    is_active: bool = True
+
+class UserCreate(UserBase):
+    password: str
+
+#Inheriting from UserBase
+class User(UserBase):
+    id: UUID = Field(default_factory=uuid4)
+    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
+    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
+
+    def get_password_hash(self):
+        return pwd_context.hash(self.password)
+    
+    #Method to verify if the provided password matches the hashed password
+    def verify_password(self, password: str):
+        return pwd_context.verify(password, self.get_password_hash())
+    
+    def to_dict(self):
+        return {
+            "id": str(self.id),
+            "username": self.username,
+            "email": self.email,
+            "is_admin": self.is_admin,
+            "is_active": self.is_active,
+            "created_at": self.created_at,
+            "updated_at": self.updated_at
+        }
+
+
+class Token(BaseModel):
+    access_token: str
+    token_type: str
+
+
+class TokenData(BaseModel):
+    username: Optional[UUID] = None
+
+
+
+#JWT token creation
+def create_access_token(data: dict, expires_delta: Optional[timedelta] = None):
```

Comment:
Models file should have only classes 
Please move all logics to another file

Reply:
done.